src/apps/backend/modules/otp/otp-service.ts

```diff
@@ -0,0 +1,34 @@
+import { PhoneNumber } from '../account/types';
+import { SMSService } from '../communication';
+
+import OtpWriter from './internal/otp-writer';
+import {
+  Otp,
+  OtpRequestError,
+} from './types';
+
+export default class OtpService {
+  public static async createOtp(
+    phoneNumber: PhoneNumber,
+  ): Promise<Otp> {
+    const otp = await OtpWriter.expirePreviousOtpAndCreateNewOtp(phoneNumber);
+
+    if (!otp) {
```

Comment:
From reading `expirePreviousOtpAndCreateNewOtp`, I don't think `otp` can be null or undefined, so this code should not exists. Please remove